src/streamlink/plugins/raiplay.py

```diff
@@ -1,45 +1,122 @@
 """
 $description Live TV channels from RAI, an Italian public, state-owned broadcaster.
 $url raiplay.it
-$type live
+$type live, vod
 $region Italy
 """
 
+import base64
 import logging
 import re
+from datetime import datetime
 from urllib.parse import parse_qsl, urlparse, urlunparse
 
-from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.exceptions import PluginError
+from streamlink.plugin import Plugin, pluginargument, pluginmatcher
 from streamlink.plugin.api import validate
 from streamlink.stream.hls import HLSStream
 from streamlink.utils.url import update_qsd
 
 
 log = logging.getLogger(__name__)
 
+_JWT_EXP_SCHEMA = validate.Schema(
+    re.compile(r"[\w-]+\.([\w-]+)\.[\w-]+"),
+    validate.get(1),
+    validate.transform(lambda s: base64.b64decode(s + "=" * (-len(s) % 4))),
+    validate.parse_json(),
+    validate.all({"exp": int}, validate.get("exp")),
```

Comment:
Unnecessary `validate.all` wrapper. `validate.Schema` already is an `all` schema.
https://streamlink.github.io/latest/api/validate.html#streamlink.plugin.api.validate.Schema